src/frontend/src/helpers/Login.ts

```diff
@@ -0,0 +1,59 @@
+import * as $ from "jquery";
+export {Login, LoginError, Connection}
+
+class LoginError extends Error {
+  constructor(msg: string,
+              public username?: string,
+              public status?: number,
+              public statusText?: number) {
+    super(msg);
+  }
+}
+
+class Login {
```

Comment:
Not sure this needs to be class -- wouldn't login(url, user, password, string, reset) be easier?